cdap-client/src/main/java/co/cask/cdap/client/config/ClientConfig.java

```diff
@@ -122,10 +123,23 @@ public URL resolveURLNoVersion(String path) throws MalformedURLException {
    * @return URL of the resolved path
    * @throws MalformedURLException
    */
+  @Deprecated
   public URL resolveNamespacedURLV3(Id.Namespace namespace, String path) throws MalformedURLException {
     return getConnectionConfig().resolveNamespacedURI(namespace, Constants.Gateway.API_VERSION_3_TOKEN, path).toURL();
```

Comment:
For deprecated method, you should delegate the call to the non-deprecated one.

```return resolveNamespacedURLV3(namespace.toEntityId(), path);```

Also, add a `@deprecate` in the javadoc to document what's the alternative.